contracts/vaults/RETHVault/base/RibbonVault.sol

```diff
@@ -0,0 +1,770 @@
+        ShareMath.assertUint104(assetBalance);
+        vaultState.lastLockedAmount = uint104(assetBalance);
+
+        vaultState.round = 1;
+    }
+
+    /**
+     * @dev Throws if called by any account other than the keeper.
+     */
+    modifier onlyKeeper() {
+        require(msg.sender == keeper, "!keeper");
+        _;
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice Sets the new keeper
+     * @param newKeeper is the address of the new keeper
+     */
+    function setNewKeeper(address newKeeper) external onlyOwner {
+        require(newKeeper != address(0), "!newKeeper");
+        keeper = newKeeper;
+    }
+
+    /**
+     * @notice Sets the new fee recipient
+     * @param newFeeRecipient is the address of the new fee recipient
+     */
+    function setFeeRecipient(address newFeeRecipient) external onlyOwner {
+        require(newFeeRecipient != address(0), "!newFeeRecipient");
+        require(newFeeRecipient != feeRecipient, "Must be new feeRecipient");
+        feeRecipient = newFeeRecipient;
+    }
+
+    /**
+     * @notice Sets the management fee for the vault
+     * @param newManagementFee is the management fee (6 decimals). ex: 2 * 10 ** 6 = 2%
+     */
+    function setManagementFee(uint256 newManagementFee) external onlyOwner {
+        require(
+            newManagementFee < 100 * Vault.FEE_MULTIPLIER,
+            "Invalid management fee"
+        );
+
+        // We are dividing annualized management fee by num weeks in a year
+        uint256 tmpManagementFee =
+            newManagementFee.mul(Vault.FEE_MULTIPLIER).div(WEEKS_PER_YEAR);
+
+        emit ManagementFeeSet(managementFee, newManagementFee);
+
+        managementFee = tmpManagementFee;
+    }
+
+    /**
+     * @notice Sets the performance fee for the vault
+     * @param newPerformanceFee is the performance fee (6 decimals). ex: 20 * 10 ** 6 = 20%
+     */
+    function setPerformanceFee(uint256 newPerformanceFee) external onlyOwner {
+        require(
+            newPerformanceFee < 100 * Vault.FEE_MULTIPLIER,
+            "Invalid performance fee"
+        );
+
+        emit PerformanceFeeSet(performanceFee, newPerformanceFee);
+
+        performanceFee = newPerformanceFee;
+    }
+
+    /**
+     * @notice Sets a new cap for deposits
+     * @param newCap is the new cap for deposits
+     */
+    function setCap(uint256 newCap) external onlyOwner {
+        require(newCap > 0, "!newCap");
+        ShareMath.assertUint104(newCap);
+        emit CapSet(vaultParams.cap, newCap);
+        vaultParams.cap = uint104(newCap);
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice Deposits ETH into the contract and mint vault shares.
+     */
+    function depositETH() external payable nonReentrant {
```

Comment:
had to create new RibbonVault (vs using from base vault) since depositETH is not `virtual`